ecs/backend.go

```diff
@@ -38,9 +39,24 @@ const backendType = store.EcsContextType
 
 // ContextParams options for creating AWS context
 type ContextParams struct {
-	Description string
-	Region      string
-	Profile     string
+	Name         string
+	Description  string
+	AccessKey    string
+	SecretKey    string
+	SessionToken string
+	Profile      string
+	Region       string
+	CredsFromEnv bool
+}
+
+func (c ContextParams) haveRequiredCredentials() bool {
+	if c.AccessKey == "" || c.SecretKey == "" {
+		return false
+	}
+	if c.Region == "" && c.Profile == "" {
+		return false
+	}
+	return true
```

Comment:
Just reading this seems confusing, you could have for example profile + secret + key defined, and no region, => true. (obviously a strange combination, but this way to group if statements seems strange. Maybe returning explicit errors here rather than a boolean will make things clearer
Also, the method should be renamed as it checks region and not only credentials